api/src/test/java/org/openmrs/api/db/hibernate/HibernateProgramWorkflowDAOTest.java

```diff
@@ -0,0 +1,789 @@
+    	Program program = hibernateProgramWorkflowDAO.getProgram(1);
+    	
+    	assertNotNull(program);
+    }
+
+	/**
+	 * @see HibernateProgramWorkflowDAO#getProgram(Integer)
+	 */
+    @Test
+	public void getProgram_shouldReturnNullIfProgramWithIdDoesNotExist() {
+		Program program = hibernateProgramWorkflowDAO.getProgram(500);
+
+		assertNull(program);
+	}
+
+	/**
+	 * @see org.openmrs.api.db.hibernate.HibernateProgramWorkflowDAO#getAllPrograms(boolean)
+	 */
+	@Test
+    public void getAllPrograms_shouldReturnAllProgramsIfIncludeRetiredFlagIsTrue() {
+    	List<Program> programList = hibernateProgramWorkflowDAO.getAllPrograms(true);
+    	
+    	assertEquals(6, programList.size());
+    }
+
+	/**
+	 * @see org.openmrs.api.db.hibernate.HibernateProgramWorkflowDAO#getAllPrograms(boolean)
+	 */
+    @Test
+	public void getAllPrograms_shouldReturnOnlyActiveProgramsIfRetiredFlagIsFalse() {
+		List<Program> programList = hibernateProgramWorkflowDAO.getAllPrograms(false);
+
+		assertEquals(3, programList.size());
+	}
+
+	/**
+	 * @see org.openmrs.api.db.hibernate.HibernateProgramWorkflowDAO#getProgramsByName(String, boolean)
+	 */
+	@Test
+    public void getProgramsByName_shouldReturnAllProgramsMatchingTheNameIfIncludeRetiredFlagIsTrue() {
+    	List<Program> programList = hibernateProgramWorkflowDAO.getProgramsByName("test-program-1", true);
+    	
+    	assertEquals(2, programList.size());
+    }
+
+	/**
+	 * @see org.openmrs.api.db.hibernate.HibernateProgramWorkflowDAO#getProgramsByName(String, boolean)
+	 */
+    @Test
+	public void getProgramsByName_shouldReturnOnlyNonRetiredProgramsMatchingTheNameIfIncludeRetiredFlagIsFalse() {
+		List<Program> programList = hibernateProgramWorkflowDAO.getProgramsByName("test-program-1", false);
+
+		assertEquals(1, programList.size());
+	}
+
+	/**
+	 * @see org.openmrs.api.db.hibernate.HibernateProgramWorkflowDAO#getProgramsByName(String, boolean)
+	 */
+	@Test
+	public void getProgramsByName_shouldReturnEmptyIfNameDoesNotMatchAnyOfThePrograms() {
+		List<Program> programList = hibernateProgramWorkflowDAO.getProgramsByName("random-program-1", true);
+
+		assertEquals(0, programList.size());
+	}
+
+	/**
+	 * @see org.openmrs.api.db.hibernate.HibernateProgramWorkflowDAO#findPrograms(java.lang.String)
+	 */
+	@Test
+    public void findPrograms_shouldReturnAllProgramsContainingTheMatchingText() {
+		List<Program> programList = hibernateProgramWorkflowDAO.findPrograms("test");
+		
+		assertEquals(3, programList.size());
+    }
+
+	/**
+	 * @see org.openmrs.api.db.hibernate.HibernateProgramWorkflowDAO#findPrograms(java.lang.String)
+	 */
+    @Test
+	public void findPrograms_shouldReturnAnEmptyListIfNoneOfTheProgramsContainTheMatchingText() {
+		List<Program> programList = hibernateProgramWorkflowDAO.findPrograms("xyz");
+
+		assertEquals(0, programList.size());
+	}
+
+	/**
+	 * @see org.openmrs.api.db.hibernate.HibernateProgramWorkflowDAO#deleteProgram(org.openmrs.Program)
+	 */
+	@Test
```

Comment:
@akshay-jha , try indenting with spaces rather than tabs. I think that might solve this. I had a similar problem.